### ISSUE TITLE
Can you please make the versioning story consistent before it's too late?

### ISSUE THREAD

**Issue by enyim** [dotnet/coreclr#7263]
So, .NET Core 1.0.1 is released, but some parts of it are called 1.0.4

It is the same as in the early .NET versions where .NET vs CLR were differently versioned (and confusing people left and right).

It'd be much friendlier to have everything the same version.

**Comment by benaadams** [comment]
That's handled by the wrapped up meta package, which includes all the base libraries:
```
"Microsoft.NETCore.App": "1.0.0
"Microsoft.NETCore.App": "1.0.1
"Microsoft.NETCore.App": "1.1.0
```

**Comment by enyim** [comment]
Sure, but

    "dependencies": {
        "Microsoft.NETCore.Runtime.CoreCLR":"1.0.4"
    },

(https://github.com/dotnet/coreclr/releases/tag/v1.0.4)

**Comment by benaadams** [comment]
That's to update the individual packages if you are using `"Microsoft.NETCore.App": "1.0.0`

`"Microsoft.NETCore.App": "1.0.1` should already contain them

**Comment by danmosemsft** [comment]
@leecow what is necessary here ? is this for 2.0? please set milestone

**Comment by leecow** [comment]
We will be snapping as much as possible to 2.0 and, as @benaadams mentions, the meta-package level will be kept consistent. We will also be snapping the CLI version to match each release so we can avoid that bit of version spaghetti. In other words, when we ship .NET Core 2.0.1 the plan is there will also be a .NET Core SDK 2.0.1.

At the package level, there will be version divergence which we can't reasonably address - eg any of the 4.0.x package versions which pre-date .NET Core builds. The trickier bit is  even if we version all possible packages at release at release time to 2.0, what do we do for patch updates? Do we bump the version numbers of all packages regardless of whether a given package has been updated? We're chewing on this one. 

Thoughts?

**Comment by benaadams** [comment]
Not bumping means better package sharing; though it depends if that is maintained (e.g. 2.1.x referring to 2.0.1 packages). No bumping makes it easier to work out what the current version of a package is "in the wild" as otherwise you are causing conflicts that's only essentially due to aliasing.

Not bumping means more difficult manual reving of packages (I've moved back to nuget updates for this reason rather than hand editing). However that should be covered by the meta-package as only one thing to update?

**Comment by leecow** [comment]
Yes, manual rev'ing should only be needed for packages not include in a referenced meta-package.

**Comment by karelz** [comment]
Do we still need this issue in CoreCLR, or is it tracked by something in CLI?

**Comment by karelz** [comment]
cc @gkhanna79

**Comment by gkhanna79** [comment]
This is by design since all packages are expected to evolve on their own. The highlevel version is always of M.N.App and is what should be used as a reference point. 

It is possible that M.N.App can increment without increment the version of all constituent packages, so having all package versions to be the same is not always possible.

**Comment by karelz** [comment]
Should we close it then? Or is it tracking anything useful? @leecow?

IMO if we plan to take any action, we should take it in 2.0, or just close it as By Design.

**Comment by leecow** [comment]
Details are being worked out in https://github.com/dotnet/designs-review/issues
